### PR TITLE
[Engine] ChatGLM-1 & 2 Enabling

### PR DESCRIPTION
**Pull request by Zhenzhong1**
## Type of Change

Feature add

## Description

New models add:
* [ChatGLM-1](https://huggingface.co/THUDM/chatglm-6b)
* [ChatGLM-2](https://huggingface.co/THUDM/chatglm2-6b)

Modification:
* Update README.md

## Expected Behavior & Potential Risk

N/A.

## How has this PR been tested?

Manual and CI tests.

## Dependency Change?

Add the sentencepiece third party.

### CODE REVIEW

**Comment by Zhenzhong1** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/42/

File: intel_extension_for_transformers/backends/neural_engine/graph/scripts/convert_chatglm.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    Q8_0 = 8
+
+
+class ModelType(Enum):
+    CHATGLM = 1
+    CHATGLM2 = 2
+
+
+def quantize_q8_0(tensor: torch.Tensor) -> torch.CharTensor:
+    # equivalent to ne_quantize_q8_0 in ne.c
```
Comment (airMeng) [inline intel_extension_for_transformers/backends/neural_engine/graph/scripts/convert_chatglm.py:56]: we don't need quantization in scripts

File: intel_extension_for_transformers/backends/neural_engine/graph/scripts/convert_chatglm.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        model = model.merge_and_unload()
+
+    if hasattr(model.config, "multi_query_attention"):
+        ChatGLM2Converter.convert(model, tokenizer, ne_type, args.outfile)
+    else:
+        ChatGLMConverter.convert(model, tokenizer, ne_type, args.outfile)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (airMeng) [inline intel_extension_for_transformers/backends/neural_engine/graph/scripts/convert_chatglm.py:382]: format

File: intel_extension_for_transformers/backends/neural_engine/graph/application/ChatGLM/chatglm.cpp [hunk 3]
```diff
@@ -0,0 +1,1250 @@
...
+    : word_embeddings(ctx, config.vocab_size, config.hidden_size), final_layernorm(ctx, config.hidden_size) {
+    layers.reserve(config.num_hidden_layers);
+    for (int layer_id = 0; layer_id < config.num_hidden_layers; layer_id++) {
+        layers.emplace_back(ctx, config.hidden_size, config.num_attention_heads, config.num_hidden_layers,
+                            config.max_length);
+    }
+}
+
+ne_tensor *ChatGLMModel::forward(ModelContext *ctx, ne_tensor *input_ids, int n_past, int n_ctx) const {
+    ne_context *gctx = ctx->ctx_b.get();
```
Comment (airMeng) [inline intel_extension_for_transformers/backends/neural_engine/graph/application/ChatGLM/chatglm.cpp:816]: model structure should be under models/

**Review by airMeng — CHANGES_REQUESTED** [review]
please align with latest implementation(llama, gpt-j)

File: intel_extension_for_transformers/backends/neural_engine/graph/application/ChatGLM/chatglm.cpp (hunk 3 above)
Comment (Zhenzhong1) [reply intel_extension_for_transformers/backends/neural_engine/graph/application/ChatGLM/chatglm.cpp:816]: The https://github.com/intel/intel-extension-for-transformers/pull/107 PR will refactor the model.

File: intel_extension_for_transformers/backends/neural_engine/graph/scripts/convert_chatglm.py (hunk 1 above)
Comment (Zhenzhong1) [reply intel_extension_for_transformers/backends/neural_engine/graph/scripts/convert_chatglm.py:56]: https://github.com/intel/intel-extension-for-transformers/pull/107 provides a quant cpp file.

**Pull request closed without merging**